sqlglot/generator.py

```diff
@@ -208,6 +208,8 @@ class Generator:
         exp.WithJournalTableProperty: exp.Properties.Location.POST_NAME,
     }
 
+    WITH_JOIN_HINT = False
```

Comment:
the default should be true, and others should say false, that way the generic generator is more inclusive

Reply:
I think I originally suggested setting the default to `False` because only T-SQL has this concept AFAIK (basically to avoid repeating setting this flag to `False` for the other dialects).

Reply:
We have kind of an issue i think with these database engine specifics, because like with the already implemented table hints I see that the generated sql in case of transpiling T-SQL to Spark SQL gives the following input/output:

Input:
```
SELECT
  a.x,
  b.y
FROM a
INNER JOIN b WITH (SNAPSHOT)
  ON a.id = b.id
```

While `WITH (SNAPSHOT)` wont work in Spark SQL and is something T-SQL specific. Shouldnt we have something available which checks whether the read dialect is the same as the write dialect and generate these dialect specific syntax?

Reply:
spark should just not have the hint

Reply:
Indeed, thats the problem with this approach. Do you have an idea on how to make sure the hint isnt outputted for other dialects?

Reply:
Yeah, but what if another dialect also has table hints but in a whole different manner. In that situation you need to be able to identify whether the hint belongs to a specific dialect or we would need to map them somehow right?

Reply:
Ok, ill leave this for now as T-SQL is the first dialect with a join hint.

Reply:
i'll change it to true after you merge it in

Reply:
I dont understand. That would result in join hints from TSQL in for example Spark SQL?

Reply:
spark needs to say false then

Reply:
Then it would work for Spark, but not for postgresql eg.

Reply:
right, so we need to add false to every dialect that doesn't support it

Reply:
Ah right, im on track now :). Ill do so in a few.

Reply:
Ok i pushed the altered default value in the generator and created an override for the other dialects.